airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py

```diff
@@ -222,40 +242,59 @@ def clear_urls(record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
 
     @backoff_policy
     def _read_records(self, params: Mapping[str, Any]) -> Iterator:
-        return self._api.account.get_ad_creatives(params=params)
+        for account in self._api.accounts:
+            yield account.get_ad_creatives(params=params)
 
 
 class Ads(FBMarketingIncrementalStream):
     """doc: https://developers.facebook.com/docs/marketing-api/reference/adgroup"""
 
     entity_prefix = "ad"
     enable_deleted = True
+    partition_field = "account_id"
 
     @backoff_policy
     def _read_records(self, params: Mapping[str, Any]):
-        return self._api.account.get_ads(params=params, fields=[self.cursor_field])
+        _params = deep_merge({}, params)
+        filtering_by_partition = _params.pop(f"filtering_by_{self.partition_field}", None)
+
+        for account in self._api.accounts:
+            params_by_partition = deep_merge(_params, filtering_by_partition.get(account.get(self.partition_field), {}))
+            yield from account.get_ads(params=params_by_partition, fields=[self.cursor_field])
 
 
 class AdSets(FBMarketingIncrementalStream):
     """doc: https://developers.facebook.com/docs/marketing-api/reference/ad-campaign"""
 
     entity_prefix = "adset"
     enable_deleted = True
+    partition_field = "account_id"
 
     @backoff_policy
     def _read_records(self, params: Mapping[str, Any]):
-        return self._api.account.get_ad_sets(params=params)
+        _params = deep_merge({}, params)
+        filtering_by_partition = _params.pop(f"filtering_by_{self.partition_field}", None)
+
+        for account in self._api.accounts:
+            params_by_partition = deep_merge(_params, filtering_by_partition.get(account.get(self.partition_field), {}))
+            yield from account.get_ad_sets(params=params_by_partition)
 
 
 class Campaigns(FBMarketingIncrementalStream):
     """doc: https://developers.facebook.com/docs/marketing-api/reference/ad-campaign-group"""
 
     entity_prefix = "campaign"
     enable_deleted = True
+    partition_field = "account_id"
 
     @backoff_policy
     def _read_records(self, params: Mapping[str, Any]):
-        return self._api.account.get_campaigns(params=params)
+        _params = deep_merge({}, params)
+        filtering_by_partition = _params.pop(f"filtering_by_{self.partition_field}", None)
+
+        for account in self._api.accounts:
```

Comment:
I see many repetitions of this, naturally CDK use the concept of `stream slices` to express loops in read_records. We can have default stream_slices methods looks like this:
```python
def stream_slices(....):
    for account in self._api.accounts:
        yield {'account': account}
```
then 
`request_params` method will take into account slice, no need to have `partition_field`.